### PR TITLE
Remove item from inbox

### LINKED ISSUE #141: LONGER LIST: After item dragged into LONG LIST, remove from LONGER LIST
blocked by #140

### PR DESCRIPTION
**Pull request by bhavinsen**
Fixed #141

### CODE REVIEW

File: cypress/integration/podcasts.spec.js [hunk 1]
```diff
@@ -68,14 +68,22 @@ describe("newsdesk logged in", () => {
...
     cy.get("#inbox-column").should("contain", "new share item")
+    switchPodCast("dev sandbox")
+    cy.get("section[slotno=7]").should("contain", "new share item")
+    cy.get("section[slotno=7] button[title='Share to podcast']").click()
+    cy.get("section[slotno=7] input[id='dev2'][type='checkbox']").click()
+    cy.get("section[slotno=7] button[title='Share to podcast']").click()
+    cy.get("section[slotno=7]").should("contain", "new share item", {
+      force: true,
+    })
+    cy.get("section[slotno=7] button[title='Delete']").click()
```
Comment (chrisns) [inline cypress/integration/podcasts.spec.js:86]:
```suggestion
    cy.get("section[slotno=7] button[title='Delete']").click()
    switchPodCast("dev 2 sandbox")
    cy.get("#inbox-column").should("not.contain", "new share item")
```

File: cypress/integration/podcasts.spec.js [hunk 2]
```diff
@@ -102,9 +110,21 @@ describe("newsdesk logged in", () => {
...
+    cy.get("section[slotno=7] button[title='Share to podcast']").click()
+    cy.get("section[slotno=7] input[id='dev2'][type='checkbox']").click()
+    switchPodCast("dev 2 sandbox")
+    cy.get("#inbox-column").should("contain", "Remove item in inbox")
+    switchPodCast("dev sandbox")
+    cy.get("section[slotno=7] ul div")
+      .eq(0)
+      .trigger("dragend", { clientX: 820, clientY: 402 }, { force: true })
+    cy.get("section[slotno=6]").should("contain", "Remove item in inbox")
+    cy.get("section[slotno=6] button[title='Delete']").click()
```
Comment (chrisns) [inline cypress/integration/podcasts.spec.js:126]: no, this is dragging from inbox to drafts, so the code looks something more like this:
```suggestion
    cy.get("section[slotno=7] textarea").type("Remove item in inbox{enter}")
    cy.get("section[slotno=7] button[title='Share to podcast']").click()
    cy.get("section[slotno=7] input[id='dev2'][type='checkbox']").click()
    switchPodCast("dev 2 sandbox")
    cy.get("#inbox-column section[slotno=7] ul div")
      .eq(0)
      .trigger("dragend", { clientX: 820, clientY: 402 }, { force: true })
    cy.get("#inbox-column").should("not.contain", "Remove item in inbox")
```

File: src/components/molecules/Cards/ItemCard.vue [hunk 3]
```diff
@@ -22,12 +22,48 @@
...
-        <BookmarkIcon />
-      </ListActionButton>
-      <ListActionButton @click="emits('share', item, 'dev2')" title="Share to dev2">
-        <BookmarkIcon />
+      <ListActionButton title="Share to podcast">
+        <Popover as="div" class="relative pt-1 text-sm">
+          <div>
+            <PopoverButton>
+              <BookmarkIconSolid v-if="item.shared" />
+              <BookmarkIcon v-else />
```
Comment (chrisns) [inline src/components/molecules/Cards/ItemCard.vue:30]:
```suggestion
              <BookmarkIconSolid/>
```
doesn't need to toggle the item.shared status

File: src/pages/Home.vue [hunk 4]
```diff
@@ -218,8 +224,23 @@ const events = {
...
-  onClickShare(item: Item, destination: string) {
-    shareStore.sendItem(item, destination, props.podcastId)
+  onClickShare(item: Item, destination: string = "", podcastNameToShare: [],getDeletePodcastId: string = "") {
+    if(destination !== ""){
+      shareStore.sendItem(item, destination, props.podcastId)
+    }
+    if(getDeletePodcastId !== ""){
+      shareStore.deleteItem(item, getDeletePodcastId, props.podcastId)
+    }
+    if(destination == props.podcastId || getDeletePodcastId ==props.podcastId){
```
Comment (chrisns) [inline src/pages/Home.vue:234]:
```suggestion
    if(destination == props.podcastId || getDeletePodcastId == props.podcastId){
```
eslint should have picked this up

File: src/pages/Home.vue [hunk 5]
```diff
@@ -218,8 +224,23 @@ const events = {
...
+      shareStore.connect(props.podcastId)
+    }
+    item.sharePodcast = podcastNameToShare
+    if (podcastNameToShare.length > 0) {
+      item.shared = true
+    } else {
+      item.shared = false
+    }
+    itemStore.updateItem(item, props.podcastId, docname)
```
Comment (chrisns) [inline src/pages/Home.vue:243]: don't need to do this
```suggestion
```

File: src/store/itemShare.ts [hunk 6]
```diff
@@ -21,23 +23,25 @@ export const useShareStore = defineStore("share", {
...
+      this.inbox = {}
       // TODO: swap this out with a collectionGroup query
       usePodcastStore().getPodcasts.forEach((podcast: podcasts) => {
-        onSnapshot(
-          doc(db, podcastname, "inbox", podcast.id, "shares"),
-          (doc) => {
-            this.inbox[podcast.id] = doc.data()?.items ?? []
-          },
-        )
+        getDoc(
```
Comment (chrisns) [inline src/store/itemShare.ts:29]: getDoc i don't think does a subscribe to updates does it?

File: src/types/item.ts [hunk 7]
```diff
@@ -3,4 +3,5 @@ export interface Item {
   slot: 1 | 2 | 3 | 4 | 5 | 6 | 7
   shared?: boolean
   id?: string
+  sharePodcast?: []
```
Comment (chrisns) [inline src/types/item.ts:6]:
```suggestion
  id?: string
```

**Review by chrisns — CHANGES_REQUESTED** [review]

File: src/store/itemShare.ts (hunk 6 above)
Comment (bhavinsen) [reply src/store/itemShare.ts:29]: 'getDoc' is got only shared podcast items.
when we use 'onSnapshot' then all-time inbox data are updated then do not show properly inbox items.

File: src/types/item.ts (hunk 7 above)
Comment (bhavinsen) [reply src/types/item.ts]: I have removed the `shared` field but `sharePodcast` is used for which item is shared to which podcast so we can't remove this field.
Please refer -> https://i.imgur.com/71e4MRr.png

File: cypress/integration/podcasts.spec.js (hunk 2 above)
Comment (bhavinsen) [reply cypress/integration/podcasts.spec.js:164]: - Inbox dragged functionality is not implemented on this branch so your suggestion is not working for this test
- I have implemented that when an item is dragged from one slot to another slot then this item is removed from the inbox.

**Comment by mrjamieeast** [comment]
This works for me

**Comment by mrjamieeast** [comment]
This works for me

File: src/types/item.ts (hunk 7 above)
Comment (chrisns) [reply src/types/item.ts]: its a copy, theres no value in tracking the content of the share, or suggesting there is some capability to unshare it be unticking it

File: src/store/itemShare.ts (hunk 6 above)
Comment (chrisns) [reply src/store/itemShare.ts:29]: right, but the inbox needs to be constantly updated in realtime

Comment (bhavinsen) [reply src/store/itemShare.ts:29]: When an item is shared then I have re-called the 'getDoc' function so it automatically gets real-time inbox data.

Comment (chrisns) [reply src/store/itemShare.ts:29]: that is not real-time.
`user1` shares `item1` from `podcast1` to `podcast2`
`user2` looking at `podcast2` should see the `item1` appear without having to carry out any actions

**Comment by chrisns** [comment]
lets revisit this when #230 is done

**Comment by mrjamieeast** [comment]
This works for me

File: src/types/item.ts [hunk 8]
```diff
@@ -2,4 +2,5 @@ export interface Item {
   text: string
   slot: 1 | 2 | 3 | 4 | 5 | 6 | 7
   id?: string
+  sharePodcast?: []
```
Comment (chrisns) [inline src/types/item.ts:5]: this appears to have come back?

File: src/pages/Home.vue [hunk 9]
```diff
@@ -141,6 +142,19 @@ const hideShowColumn = reactive({
...
 })
 
+const draggedInbox = (x: number, y: number, text: string, key: string) => {
+  const slot = <Item["slot"]>parseInt(
+    <string>// @ts-ignore
+    document.elementFromPoint(x, y)?.closest("section")?.attributes["slotno"]?.value,
+  )
+
+  if(slot) {
+    itemStore.addItem({ text, slot, sharePodcast: [] }, props.podcastId, docname)
```
Comment (chrisns) [inline src/pages/Home.vue:152]: this appears to have come back?

**Review by chrisns — CHANGES_REQUESTED** [review]
looks like your conflict resolution didn't work

File: src/types/item.ts (hunk 8 above)
Comment (bhavinsen) [reply src/types/item.ts]: I have removed `sharePodcast`.

File: src/pages/Home.vue (hunk 9 above)
Comment (bhavinsen) [reply src/pages/Home.vue]: I have removed `sharePodcast`.

File: src/pages/Home.vue (hunk 5 above)
Comment (bhavinsen) [reply src/pages/Home.vue]: I have already removed it.

File: src/pages/Home.vue (hunk 4 above)
Comment (bhavinsen) [reply src/pages/Home.vue]: I have already removed it.

**Comment by ArindamReact** [comment]
approved

File: src/pages/Home.vue [hunk 10]
```diff
@@ -160,6 +161,18 @@ const hideShowColumn = reactive({
...
+const draggedInbox = (x: number, y: number, text: string, key: string) => {
+  const slot = <Item["slot"]>parseInt(
+    <string>// @ts-ignore
+    document.elementFromPoint(x, y)?.closest("section")?.attributes["slotno"]?.value,
+  )
+
+  if(slot) {
+    itemStore.addItem({ text, slot }, props.podcastId, docname)
+    shareStore.removeDraggedItem(text, props.podcastId, key)
+    shareStore.connect(props.podcastId)
```
Comment (chrisns) [inline src/pages/Home.vue:173]: do we still need this if its been subscribed?

**Review by ArindamReact — DISMISSED** [review]

**Review by ArindamReact — APPROVED** [review]

**Review by ArindamReact — APPROVED** [review]

**Review by ArindamReact — APPROVED** [review]

**Review by ArindamReact — APPROVED** [review]

**Review by ArindamReact — DISMISSED** [review]
up-to-date

**Review by ArindamReact — APPROVED** [review]

**Review by ArindamReact — APPROVED** [review]

**Review by ArindamReact — DISMISSED** [review]

**Review by ArindamReact — APPROVED** [review]

**Comment by ArindamReact** [comment]
All conversations are resolved

**Comment by ArindamReact** [comment]
All conversations are resolved

**Pull request closed without merging**